config/hiveadmission/clusterdeployment-webhook.yaml

```diff
@@ -20,5 +20,5 @@ webhooks:
     apiVersions:
     - v1alpha1
     resources:
-    - clusterdeployments
+    - clusterdeploymentvalidators
```

Comment:
I don't think that this is the right thing to change. This is the resource that the validator is validating. What should change is the name of the admissionregistration, the name of the webhook, and the path of the service.

Reply:
@staebler I can see that you have sent a PR to add validator for clusterprovisions. Does this comment still holds?

Reply:
Yes. This comment is not related to that PR.

Reply:
The `rules` field stipulates which resources the validating webhook act upon. With your change, you are setting up the webhook to validate `clusterdeploymentvalidators` resources in `hive.openshift.io/v1alpha1`. The webhook should be validating `clusterdeployments` resources.

Reply:
> Rules describes what operations on what resources/subresources the webhook cares about. The webhook cares about an operation if it matches _any_ Rule.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.16/#validatingwebhook-v1beta1-admissionregistration-k8s-io

Reply:
It took sometime for me to understand what you are saying. Sending an update to the PR with the fix.